src/openrct2/entity/Guest.cpp

```diff
@@ -4513,16 +4513,26 @@ void Guest::UpdateRideApproachExitWaypoints()
             return;
         }
 
-        const auto& rtd = ride->GetRideTypeDescriptor();
-        CoordsXY targetLoc = rtd.GetGuestWaypointLocation(*vehicle, *ride, CurrentRideStation);
-
         RideObjectEntry* rideEntry = vehicle->GetRideEntry();
-        CarEntry* carEntry = &rideEntry->Cars[vehicle->vehicle_type];
+        if (rideEntry == nullptr)
+            return;
+
+        if (vehicle->vehicle_type >= std::size(rideEntry->Cars))
+            return;
 
-        Guard::Assert((Var37 & 3) < 3);
-        targetLoc.x += carEntry->peep_loading_waypoints[Var37 / 4][Var37 & 3].x;
-        targetLoc.y += carEntry->peep_loading_waypoints[Var37 / 4][Var37 & 3].y;
+        const CarEntry& carEntry = rideEntry->Cars[vehicle->vehicle_type];
 
+        const size_t fours = Var37 / 4;
```

Comment:
I think we could name this `carPosition` or `carNumber`

Reply:
```suggestion
        const size_t carPosition = Var37 / 4;
```